block/src/Pos/Block/Network/Retrieval.hs

```diff
@@ -110,27 +109,26 @@ retrievalWorkerImpl keepAliveTimer SendActions {..} =
         slotDuration <- fromIntegral . toMicroseconds <$> getCurrentEpochSlotDuration
         setTimerDuration keepAliveTimer $ 3 * slotDuration
         startTimer keepAliveTimer
+        -- Exception handlers are installed locally, on the 'thingToDoNext',
+        -- to ensure that network troubles, for instance, do not kill the
+        -- worker.
         thingToDoNext
         mainLoop
-    mainLoopE e = do
-        -- REPORT:ERROR 'reportOrLogE' in block retrieval worker.
-        reportOrLogE "retrievalWorker mainLoopE: error caught " e
-        delay (sec 1)
```

Comment:
Was this a very important thing to do? I can't imagine why.

Reply:
@volhovm added it, I'd like him to comment on the removal.

Reply:
Because if there's an error in retrieval worker that reproduces on every iteration, the worker will panic, writing warning logs every `n` ms, which is terrible. I think it's a good heuristic to pause in any worker handler.

Reply:
As a general principle @volhovm is right to worry about that problem. In this case we can see that we do not need to do anything special to avoid the "repeated failure" problem because each time it fails it has already taken one or more items from the block request queue, so at worst it can fail once for a whole queue worth of entries (since we process all queue entries and then act on them). So there's no need to do anything special here.

Reply:
If the queue is empty, but `tryReadTMVar recHeaderVar` is `Just`, then it's possible to loop forever.

Reply:
But, if that's the outcome of the STM transaction, then we're going to `handleRecoveryWithHandler`, which will drop the header from the `TMVar` in case of exception, and eventually (by way of `getProcessBlocks`) drop it if it ultimately receives a more-difficult block... it's a little convoluted.